### PR TITLE
Java: Add getAllowedActions and tests

### PR DESCRIPTION
**Pull request by joshrotenberg**
See #746.

PR checklist:

- [ ] Added changelog entry.

### CODE REVIEW

**Comment by joshrotenberg** [comment]
FYI, I get the following warnings when I run the tests:

```
[warning] Singleton variable actor is unused or undefined, see <https://docs.osohq.com/using/polar-syntax.html#variables>
001: allow(actor: Actor{name: "sally"}, action, resource: Widget{id: 1}) if action in ["CREATE", "READ"];
           ^
[warning] Singleton variable resource is unused or undefined, see <https://docs.osohq.com/using/polar-syntax.html#variables>
001: allow(actor: Actor{name: "sally"}, action, resource: Widget{id: 1}) if action in ["CREATE", "READ"];
                                                ^
[warning] Singleton variable actor is unused or undefined, see <https://docs.osohq.com/using/polar-syntax.html#variables>
001: allow(actor: Actor{name: "fred"}, action, resource: Widget{id: 2}) if action in [1, 2, 3, 4];
           ^
[warning] Singleton variable resource is unused or undefined, see <https://docs.osohq.com/using/polar-syntax.html#variables>
001: allow(actor: Actor{name: "fred"}, action, resource: Widget{id: 2}) if action in [1, 2, 3, 4];

```

**Comment by samscott89** [comment]
This looks great! Thanks for doing this!

Do you think it would be possible to support the wildcard variant if a variable is returned?

File: languages/java/oso/src/test/java/com/osohq/oso/OsoTest.java [hunk 1]
```diff
@@ -112,4 +112,42 @@ public void testAllowModel() throws Exception {
...
+
+  @Test
+  public void getGetAllowedActions() throws Exception {
+
+    Oso o = new Oso();
+    o.registerClass(Actor.class, "Actor");
+    o.registerClass(Widget.class, "Widget");
+
+    o.loadStr(
+        "allow(actor: Actor{name: \"sally\"}, action, resource: Widget{id: 1})"
```
Comment (samscott89) [inline languages/java/oso/src/test/java/com/osohq/oso/OsoTest.java:124]:
```suggestion
        "allow(_actor: Actor{name: \"sally\"}, action, _resource: Widget{id: 1})"
```

File: languages/java/oso/src/test/java/com/osohq/oso/OsoTest.java [hunk 2]
```diff
@@ -112,4 +112,42 @@ public void testAllowModel() throws Exception {
...
+    Actor actor = new Actor("sally");
+    Widget widget = new Widget(1);
+    HashSet<Object> actions = o.getAllowedActions(actor, widget);
+
+    assertEquals(actions.size(), 2);
+    assertTrue(actions.contains("CREATE"));
+    assertTrue(actions.contains("READ"));
+
+    o.loadStr(
+        "allow(actor: Actor{name: \"fred\"}, action, resource: Widget{id: 2})"
```
Comment (samscott89) [inline languages/java/oso/src/test/java/com/osohq/oso/OsoTest.java:136]:
```suggestion
        "allow(_actor: Actor{name: \"fred\"}, action, _resource: Widget{id: 2})"
```

**Comment by joshrotenberg** [comment]
I’ll look into it.

**Comment by joshrotenberg** [comment]
> Do you think it would be possible to support the wildcard variant if a variable is returned?

Just did a copycat of the Python version. Feedback/admonishment welcome.

File: languages/java/oso/src/main/java/com/osohq/oso/Oso.java [hunk 3]
```diff
@@ -27,6 +29,70 @@ public boolean isAllowed(Object actor, Object action, Object resource)
...
+   * @param actor the actor performing the request
+   * @param resource the resource being accessed
+   * @return HashSet<Object>
+   * @throws Exceptions.OsoException
+   */
+  public HashSet<Object> getAllowedActions(Object actor, Object resource)
+      throws Exceptions.OsoException {
+    return queryRule("allow", actor, new Variable("action"), resource).results().stream()
+        .map(action -> action.get("action"))
+        .collect(Collectors.toCollection(HashSet::new));
```
Comment (samscott89) [inline languages/java/oso/src/main/java/com/osohq/oso/Oso.java:51]:
```suggestion
    return getAllowedActions(actor, resource, false);
```

**Comment by samscott89** [comment]
> 
> 
> > Do you think it would be possible to support the wildcard variant if a variable is returned?
> 
> Just did a copycat of the Python version. Feedback/admonishment welcome.

It looks great! One tiny suggestion and this looks really good to me. Thank you again for adding this!

**Review by samscott89 — APPROVED** [review]
Looks good!

**Pull request merged**